src/arduino.c

```diff
@@ -37,6 +37,45 @@
 #include "stk500.h"
 #include "arduino.h"
 
+static int arduino_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
+  const char *extended_param;
+  int attempts;
+  int rv = 0;
+
+  for (LNODEID ln = lfirst(extparms); ln; ln = lnext(ln)) {
+    extended_param = ldata(ln);
+
+    if (sscanf(extended_param, "attempts=%2d", &attempts) == 1) {
+      PDATA(pgm)->retry_attempts = attempts;
+      pmsg_info("setting number of retry attempts to %d\n", attempts);
+      continue;
+    }
+
+    if(str_starts(extended_param, "no_autoreset")) {
+      if(!str_eq(extended_param, "no_autoreset")) {
+        pmsg_error("invalid -xno_autoreset value %s. Use -xno_autoreset\n", extended_param);
+        rv = -1;
+        break;
+      }
```

Comment:
Why not replace these 6 lines with one
```
  if(str_eq(extended_param, "no_autoreset")) {
```
Note the `str_eq()` instead of the `str_starts()`. Anything that starts with `no_autoreset` but isn't `no_autoreset` is caught later on anyway (invalid parameter).